### PR TITLE
Add get window title function [help needed]

### PR DESCRIPTION
**Pull request by Bayemite**
This is addressing issue #1448, adding a function that queries the window title.
The implementation prefers calls to platform specific functions where available.

TODO:
It would be nice if someone could help out with the Cocoa and Wayland implementations- I have no access to these platforms.

### CODE REVIEW

File: src/win32_window.c [hunk 1]
```diff
@@ -1462,6 +1462,41 @@ void _glfwPlatformDestroyWindow(_GLFWwindow* window)
         DestroyIcon(window->win32.smallIcon);
 }
 
+char* _glfwPlatformGetWindowTitle(_GLFWwindow* window)
+{
+    int count = GetWindowTextLengthW(window->win32.handle);
+    if(count == 0)
+    {
+        int error;
+        SetLastError(0);
```
Comment (intrigus) [inline src/win32_window.c:1471]: Shouldn't this be before the call to `GetWindowTextLengthW`?

Comment (Bayemite) [reply src/win32_window.c:1471]: Yes, it should be that way...
I was probably taking the documentation too literally, even when it made no sense.

**Comment by ws909** [comment]
Why was the Cocoa implementation removed?

**Comment by dougbinks** [comment]
I have started work on a revised version of this PR, which will retain author credits for @Bayemite.

Currently I have rebased the code and am in progress with switching from the old style platform functions to the new style before testing and reviewing the code on each platform.

**Pull request closed without merging**